core/src/main/java/com/amplitude/core/utilities/HttpClient.kt

```diff
@@ -111,13 +111,13 @@ internal class HttpClient(
 abstract class Connection(
     val connection: HttpURLConnection,
     val inputStream: InputStream?,
-    val outputStream: OutputStream?
+    val outputStream: OutputStream?,
 ) : Closeable {
-
     private lateinit var apiKey: String
     private lateinit var clientUploadTime: String
     private lateinit var events: String
     private var minIdLength: Int? = null
+    private var diagnostics: String? = null
```

Comment:
Can we make the diagnostics class a typed class? So we know what all diagnostics we are capturing from within the SDK.

Reply:
So you mean we pass the diagnostics object here? I guess in HTTP client we could possibly keep the string, while if we could put it in while amplitude client level